vdr/api/v1/types.go

```diff
@@ -25,3 +25,63 @@ type DIDDocument = did.Document
 
 // DIDDocumentMetadata is an alias
 type DIDDocumentMetadata = types.DocumentMetadata
+
+// DIDCreateRequest defines model for DIDCreateRequest.
+type DIDCreateRequest struct {
+	VerificationMethodRelationship
+
+	// List of DIDs that can control the new DID Document. If selfControl = true and controllers is not empty,
+	// the newly generated DID will be added to the list of controllers.
+	Controllers *[]string `json:"controllers,omitempty"`
+
+	// whether the generated DID Document can be altered with its own capabilityInvocation key.
+	SelfControl *bool `json:"selfControl,omitempty"`
+}
+
+// VerificationMethodRelationship defines model for VerificationMethodRelationship.
+type VerificationMethodRelationship struct {
+	// indicates if the generated key pair can be used for assertions.
+	AssertionMethod *bool `json:"assertionMethod,omitempty"`
+
+	// indicates if the generated key pair can be used for authentication.
+	Authentication *bool `json:"authentication,omitempty"`
```

Comment:
I think it is better not to use pointers for absent (configuration) values where possible. If AssertMethod was just a `bool`, then it would document the default of `false` on the fly. Authentication would have to be renamed to NoAuthentication as such.

No pointers also prevents the ugly `&varFalse` construction above, plus it eliminates the need for a separate withKeyFlag function—just inline the if/else in  ToFlags. ✨

Reply:
There because of ugly OpenAPI code generator, but fixed ugly/dangerous bool ptr reusage.